harness/determined/exec/harness.py

```diff
@@ -114,11 +115,31 @@ def main(train_entrypoint: str) -> int:
 
             # Step 5: Create a TrialController and execute training
             logging.info(f"Creating {controller_class.__name__} with {trial_class.__name__}.")
-            controller = controller_class.from_trial(
-                trial_inst=trial_inst,
-                context=trial_context,
-                env=env,
-            )
+            if issubclass(type(trial_class), PyTorchTrial):
+                trial_context = PyTorchTrialContext(hparams=env.hparams,
```

Comment:
break this `if` block into its own helper function?

Reply:
or even better: move the `trial_class = load.trial_class_from_entrypoint(train_entrypoint)` above the creation of `env`, and if it's pytorch trial, break out right then and never even create the `env` object.  That way your code can't be tainted by it.

Reply:
yeah, breaking out of this function before my long-winded comment about "we shouldn't use this in the new APIs we build" is perfect :)

Reply:
i tried this. but i don't know if it's better... you end up with a helper method signature of something like:
```
def create_pytorch_trial_controller_and_context(min_checkpoint_period: int,
                                                min_validation_period: int,
                                                searcher_metric_name: str,
                                                average_training_metrics: bool,
                                                checkpoint_policy: str,
                                                smaller_is_better: bool,
                                                steps_completed: Optional[int] = None,
                                                latest_checkpoint: Optional[str] = None,
                                                test_mode: Optional[bool] = False,
                                                records_per_epoch: Optional[int] = None,
                                                hparams: Optional[Dict] = None,
                                                slots_per_trial: Optional[int] = 0,
                                                num_gpus: Optional[int] = 0,
                                                core_context: det.core.Context = None,
                                                exp_conf: Optional[Dict] = None,
                                                aggregation_frequency: Optional[int] = 1,
                                                fp16_compression: Optional[bool] = False,
                                                average_aggregated_gradients: Optional[bool] = False
                                                ):
```

could pass in the cluster info object instead, but is that really better than creating the envcontext object?

Reply:
no longer applies